index.html

```diff
@@ -5058,39 +5068,47 @@ <h3 id="playback-AudioBufferSourceNode">
-            whether output of the buffer contents has begun or not.
-            </li>
-            <li>For each block of audio to be rendered:
+            <li>For each quantum of audio to be rendered:
               <ol>
-                <li>Let <code>currentTime</code> be the value of
-                <code>currentTime</code> attribute of the node's
-                <a>AudioContext</a>.
-                </li>
-                <li>Let <code>sampleRate</code> be the value of the
-                <code>sampleRate</code> attribute of the node's
-                <a>AudioContext</a>.
-                </li>
-                <li>Let <code>dt</code> be <code>1 / sampleRate</code>.
-                </li>
-                <li>Let <code>index</code> be 0.
-                </li>
-                <li>Let <code>enteredLoop</code> be <code>false</code>.
+                <li>Initialize state for this render quantum:
+                  <ol>
+                    <li>Let <code>currentTime</code> be the value of
+                    <code>currentTime</code> attribute of the node's
+                    <a>AudioContext</a>.
+                    </li>
+                    <li>Let <code>sampleRate</code> be the value of the <code>
+                      sampleRate</code> attribute of the node's
+                      <a>AudioContext</a>.
+                    </li>
+                    <li>Let <code>dt</code> be <code>1 / sampleRate</code>.
+                    </li>
+                    <li>Let <code>index</code> be 0.
+                    </li>
+                    <li>Let <code>computedPlaybackRate</code> be
+                    <code>playbackRate * pow(2, detune / 1200)</code> . This
+                    combines the two k-rate parameters affecting playback rate
+                    into a single computed quantity.
+                    </li>
+                  </ol>
                 </li>
-                <li>Let <code>computedPlaybackRate</code> be <code>playbackRate
-                * pow(2, detune / 1200)</code> . This combines the two k-rate
-                parameters affecting playback rate into a single computed
-                quantity.
+                <li>If <code>loop</code> is true,
+                  <ol>
+                    <li>If <code>loopStart</code> &gt; 0 and
+                    <code>loopEnd</code> &gt; 0 and <code>loopStart</code> &lt;
+                    <code>loopEnd</code>, let <code>actualLoopStart</code>
+                    equal <code>loopStart</code> and <code>actualLoopEnd</code>
+                    equal <code>min(loopEnd, buffer.duration)</code>.
+                    </li>
+                    <li>Else, let <code>actualLoopStart</code> equal
+                    <code>0</code> and <code>actualLoopEnd</code> equal <code>
+                      buffer.duration</code>.
+                    </li>
+                  </ol>
                 </li>
                 <li>While <code>index</code> is less than the length of the
```

Comment:
Assuming we get rid of the render quantum stuff, I think this loop should be 

While currentTime < stop do the following...

The loop body will have to update stop depending on whether stop() was called or not while rendering.